test/jdk/java/net/httpclient/Response1xxTest.java

```diff
@@ -0,0 +1,498 @@
+     * Tests that when a request is issued with a specific request timeout and the server
+     * responds with intermediate 1xx response code but doesn't respond with a final response within
+     * the timeout duration, then the application fails with a request timeout
+     */
+    @Test
+    public void test1xxRequestTimeout() throws Exception {
+        final HttpClient client = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_2)
+                .proxy(HttpClient.Builder.NO_PROXY).build();
+        TRACKER.track(client);
+        final URI requestURI = new URI(http2RequestURIBase + "/only-informational");
+        final Duration requestTimeout = Duration.ofSeconds(2);
+        final HttpRequest request = HttpRequest.newBuilder(requestURI).timeout(requestTimeout)
+                .build();
+        System.out.println("Issuing request to " + requestURI);
+        // we expect the request to timeout
+        Assert.assertThrows(HttpTimeoutException.class, () -> {
+            client.send(request, HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8));
+        });
+    }
+
+    /**
+     * Tests that when the HTTP/1.1 server sends a 101 response when the request hasn't asked
+     * for an "Upgrade" then the request fails.
+     */
+    @Test
+    public void testHTTP11Unexpected101() throws Exception {
+        final HttpClient client = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_1_1)
+                .proxy(HttpClient.Builder.NO_PROXY).build();
+        TRACKER.track(client);
+        final URI requestURI = new URI(http1RequestURIBase + "/test/bye");
+        final HttpRequest request = HttpRequest.newBuilder(requestURI).build();
+        System.out.println("Issuing request to " + requestURI);
+        // we expect the request to fail because the server sent an unexpected 101
+        Assert.assertThrows(ProtocolException.class,
+                () -> client.send(request, HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8)));
+    }
+
+
+    /**
+     * Tests that when the HTTP2 server (over HTTPS) sends a 101 response when the request
+     * hasn't asked for an "Upgrade" then the request fails.
+     */
+    @Test
+    public void testSecureHTTP2Unexpected101() throws Exception {
+        final HttpClient client = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_2)
+                .sslContext(sslContext)
+                .proxy(HttpClient.Builder.NO_PROXY).build();
+        TRACKER.track(client);
+        final URI requestURI = new URI(https2RequestURIBase + "/101");
+        final HttpRequest request = HttpRequest.newBuilder(requestURI).build();
+        System.out.println("Issuing request to " + requestURI);
+        // we expect the request to fail because the server sent an unexpected 101
+        // to check - we get here an IOException not a ProtocalException (compared to 17 or 21)
+        //Assert.assertThrows(ProtocolException.class,
```

Comment:
Typo: `ProtocalException`. I think a better comment would be:
Backport note: Later JDK versions throw a ProtocolException.

Reply:
Hi Martin, I changed the comment .